### PR TITLE
Validating Webhooks

### PR DESCRIPTION
**Pull request by goern**
implements #66

### CODE REVIEW

**Review by codificat — COMMENTED** [review]
Thanks for this!

I only have some comments/questions for now.

It LGTM, but I am not done digesting all of the changes. Among other things, I can't get this version to run either on kind (image not found, despite `kind load` working) or on OCP (secret `webhook-server-cert` not found) yet.

File: README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ curl -sSLo envtest-bins.tar.gz "https://go.kubebuilder.io/test-tools/${K8S_VERSI
 sudo mkdir /usr/local/kubebuilder
 sudo chown $(whoami) /usr/local/kubebuilder
 tar -C /usr/local/kubebuilder --strip-components=1 -zvxf envtest-bins.tar.gz
-make test SKIP_FETCH_TOOLS=1 KUBEBUILDER_ASSETS=/usr/local/kubebuilder
+make test SKIP_FETCH_TOOLS=1 KUBEBUILDER_ASSETS=/usr/local/kubebuilder ENABLE_WEBHOOKS=false
```
[the remark this answers is not in the document]
Comment (goern) [reply README.md:74]: ja, it seems to be an ENV just for `testenv`

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/aicoe/meteor-operator
+  newName: quay.io/goern/meteor-operator
```
[the remark this answers is not in the document]
Comment (goern) [reply config/manager/kustomization.yaml:15]: hmm, ja, right. that string might have been auto generated by `make deploy IMG=../goern/...`

File: main.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ import (
 	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 
 	meteorv1alpha1 "github.com/aicoe/meteor-operator/api/v1alpha1"
-	cnbi "github.com/aicoe/meteor-operator/controllers/cnbi"
+	"github.com/aicoe/meteor-operator/controllers"
```
[the remark this answers is not in the document]
Comment (goern) [reply main.go:37]: I think the scope of vars of suite_test.go is per module, so having a suite_test for all controllers will not expose k8client to the submodules (such as meteor/). We could have a suite_test.go for each module===controller, but that feels against the 'suite idea'?!

File: config/default/kustomization.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Adds namespace to all resources.
-#namespace: aicoe-meteor
+namespace: meteor-operator-system
```
[the remark this answers is not in the document]
Comment (goern) [reply config/default/kustomization.yaml:2]: I have no hard feelings on this. my rational was 'meteor is more than just cnbi' and 'i like the meteor brand' ,)

[the remark this answers is not in the document]
Comment (codificat) [reply config/default/kustomization.yaml:2]: ok, makes sense.

I still don't quite like the full `meteor-operator-system` for the namespace, but we can fine tune later

File: config/manager/kustomization.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (codificat) [reply config/manager/kustomization.yaml:15]: ok, let's talk later about where to maintain it

File: main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (codificat) [reply main.go:37]: I honestly don't know... it works fine this way, so let's keep it for now

**Comment by codificat** [comment]
Ok, I finally managed to get this working on an OCP cluster, and verified with the current sample CR:

```
$ kubectl apply -f config/samples/meteor_v1alpha1_customnbimage.yaml 
customnbimage.meteor.zone/ubi8-py38-sample-3 created
customnbimage.meteor.zone/s2i-minimal-py38-notebook created
The CustomNBImage "ubi8-py38-sample-2" is invalid: 
* metadata.annotations[opendatahub.io/notebook-image-name]: Required value: annotation is required
* metadata.annotations[opendatahub.io/notebook-image-desc]: Required value: annotation is required
* metadata.annotations[opendatahub.io/notebook-image-creator]: Required value: annotation is required

$ tkn pr ls
NAME                                    STARTED        DURATION   STATUS
cnbi-ubi8-py38-sample-3-prepare         1 minute ago   ---        Running
cnbi-s2i-minimal-py38-notebook-import   1 minute ago   ---        Running
```

and after a while:

```
$ tkn pr ls
NAME                                    STARTED         DURATION   STATUS
cnbi-ubi8-py38-sample-3-prepare         2 minutes ago   1 minute   Succeeded
cnbi-s2i-minimal-py38-notebook-import   2 minutes ago   ---        Running

$ tkn pr ls
NAME                                    STARTED         DURATION    STATUS
cnbi-ubi8-py38-sample-3-prepare         7 minutes ago   1 minute    Succeeded
cnbi-s2i-minimal-py38-notebook-import   8 minutes ago   4 minutes   Succeeded
```

Let's go!

/approve
/lgtm

**Pull request merged**